properties/types.go

```diff
@@ -0,0 +1,127 @@
+// It returns "UNSPECIFIED" for Unspecified, "INBOUND" for Inbound, and "OUTBOUND" for Outbound.
+// If the enum value doesn't match any of the predefined values, it defaults to "UNSPECIFIED".
+func (t EnvoyTrafficDirection) String() string {
+	switch t {
+	case Unspecified:
+		return "UNSPECIFIED"
+	case Inbound:
+		return "INBOUND"
+	case Outbound:
+		return "OUTBOUND"
+	}
+	return "UNSPECIFIED"
+}
+
+// EnvoyLocality identifies location of where either Envoy runs or where upstream hosts run.
+//
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#config-core-v3-locality
+type EnvoyLocality struct {
+	Region  string
+	Zone    string
+	Subzone string
+}
+
+// EnvoyExtension holds version and identification for an Envoy extension.
+//
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#config-core-v3-extension
+type EnvoyExtension struct {
+	Name     string
+	Category string
+	TypeUrls []string
+}
+
+// IstioFilterMetadata provides additional inputs to filters based on matched listeners,
+// filter chains, routes and endpoints. It is structured as a map, usually from
+// filter name (in reverse DNS format) to metadata specific to the filter. Metadata
+// key-values for a filter are merged as connection and request handling occurs,
+// with later values for the same key overriding earlier values.
+//
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#config-core-v3-metadata
+type IstioFilterMetadata struct {
+	Config   string
+	Services []IstioService
+}
+
+// IstioService holds information of the host, name and namespace of an Istio Service.
+type IstioService struct {
+	Host      string
+	Name      string
+	Namespace string
+}
+
+// IstioProxyStatsMatcher holds proxy stats name matches for stats creation. Note this is in addition to the minimum Envoy stats that
+// Istio generates by default.
+//
+// https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#ProxyConfig-ProxyStatsMatcher
+type IstioProxyStatsMatcher struct {
+	InclusionPrefixes []string
+	InclusionRegexps  []string
+	InclusionSuffixes []string
+}
+
+// IstioTrafficInterceptionMode indicates how traffic to/from the workload is captured and sent to Envoy. This
+// should not be confused with the CaptureMode in the API that indicates how the user wants traffic to be
+// intercepted for the listener. IstioTrafficInterceptionMode is always derived from the Proxy metadata.
+//
+// https://pkg.go.dev/istio.io/istio/pilot/pkg/model#TrafficInterceptionMode
+type IstioTrafficInterceptionMode int
+
+const (
+	// None indicates that the workload is not using IPtables for traffic interception.
+	None IstioTrafficInterceptionMode = iota
+	// Tproxy implies traffic intercepted by IPtables with TPROXY mode.
+	Tproxy
+	// Redirect implies traffic intercepted by IPtables with REDIRECT mode. This is our default mode.
+	Redirect
+)
```

Comment:
```suggestion
type IstioTrafficInterceptionMode int

const (
	// IstioTrafficInterceptionModeNone indicates that the workload is not using IPtables for traffic interception.
	IstioTrafficInterceptionModeNone IstioTrafficInterceptionMode = iota
	// IstioTrafficInterceptionModeTproxy implies traffic intercepted by IPtables with TPROXY mode.
	IstioTrafficInterceptionModeTproxy
	// IstioTrafficInterceptionModeRedirect implies traffic intercepted by IPtables with REDIRECT mode. This is our default mode.
	IstioTrafficInterceptionModeRedirect
)
```

Reply:
The most idiomatic Go comments are those that are clear, concise, and directly above the item they're describing. The Go community values clarity and simplicity, and the official Go documentation often uses the style seen in the first example.

Therefore, the first example is the most idiomatic:

```golang
const (
	// None indicates that the workload is not using IPtables for traffic interception.
	None IstioTrafficInterceptionMode = iota
	// Tproxy implies traffic intercepted by IPtables with TPROXY mode.
	Tproxy
	// Redirect implies traffic intercepted by IPtables with REDIRECT mode. This is our default mode.
	Redirect
)
``` 

The comments are directly above the constants they describe, and they don't repeat the constant's name, which is redundant and not necessary. The other examples are more verbose and less idiomatic.

Online examples can be found in plenty upstream well adopted modules. An example of the `net/http` package

https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/net/http/server.go;l=2873

PS: `IstioTrafficInterceptionModeRedirect` does not make any sense, as this string would not even be useful in code nor documents (_where does the type name start and end?_). `IstioTrafficInterceptionMode.Redirect` (with the dot) would make a bit more sense, but is not as golang idiomatic as mentioned in the initial argument.